openapi/src/parts/controlled_gcp_big_query_dataset.yaml

```diff
@@ -73,6 +73,28 @@ paths:
           $ref: '#/components/responses/NotFound'
         '500':
           $ref: '#/components/responses/ServerError'
+
+  /api/workspaces/v1/{workspaceId}/resources/controlled/gcp/bqdatasets/generateName:
+    parameters:
+    - $ref: '#/components/parameters/WorkspaceId'
+    post:
+      summary: Generate a cloud native controlled Gcp bucket name
+      operationId: getCloudNameFromBigQueryDatasetName
```

Comment:
generateBigQueryDatasetCloudId or name

Is there a discussion I miss about id vs name?